### DISCUSSION TITLE
Mac m1 issues

### DISCUSSION THREAD

**Discussion by TeoBale** [nvim-treesitter/nvim-treesitter#4184] [Q&A]
Hi, I have been trying to install tree-sitter on my macbook air m1 machine for a while now, I have been following the repository instructions and some online guides step by step and managed to fix most of the errors.

Right now the only remaining problem is with the parser which is compiled for x86_64 while I own an arm64 architecture

Does anyone know how to help me? Unfortunately I apologize for my ignorance but I am not very well versed in these topics

**Comment by theHamsta** [comment]
We had this problem before. One of the main reasons was that users were either using a x86_64 Neovim or a x86_64 terminal environment and so the default terminal environment was x86_64 so that the compiler decided to use that. What does `:checkhealth nvim-treesitter` print?

You can set environment variables and extra_cmd_args for cc instead of curl to control the target architecture https://github.com/nvim-treesitter/nvim-treesitter#i-want-to-use-a-http-proxy-for-downloading-the-parsers
